src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -1,10 +1,24 @@
 package com.hibob.academy.service
 
+import com.hibob.academy.dao.OwnerDao
 import com.hibob.academy.dao.PetDao
+import com.hibob.academy.entity.Owner
+import com.hibob.academy.entity.Pet
+import jakarta.ws.rs.BadRequestException
 import org.springframework.stereotype.Service
 
 @Service
 class PetService(
-    private val petDao: PetDao
+    private val petDao: PetDao,
 ) {
+
+    fun assignOwnerToPet(petId: Long, ownerId: Long) : Boolean {
+
+        return petDao.assignOwnerToPet(petId, ownerId)
+    }
+
+    fun getOwnerByPetId(petId: Long): Owner? {
+
+        return petDao.getOwnerByPetId(petId) ?: throw BadRequestException("Pet with ID $petId not found")
```

Comment:
Must every pet have an owner?
If so - the error is ok
If not - maybe we should just return null 
Was there a definition of this?